src/main/java/mate/academy/spring/dao/RoleDao.java

```diff
@@ -0,0 +1,9 @@
+package mate.academy.spring.dao;
+
+import mate.academy.spring.model.Role;
+
+public interface RoleDao {
+    Role add(Role role);
+
+    Role getByName(String roleName);
```

Comment:
[checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-spring/security-part-2/jv-spring-security-checklist)
11. Be careful in getByName() method on dao layer! You should pass enum to query (not String), otherwise you won’t be able to get Role by name.

Reply:
I don't understand how is it possible to pass Enum. Hibernate throws an error and it seems logical. When passing a String everything works just fine